channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -1010,9 +1021,11 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControls {
                         _self._buffer.markAsSent(payload);
                     }
 
+                    let result: void | IPromise<boolean>;
                     _sendPostMgr.preparePayload((processedPayload: IPayloadData) => {
-                        return sendPostFunc(processedPayload, onComplete, !isAsync);
+                        result = sendPostFunc(processedPayload, onComplete, !isAsync);
```

Comment:
This won't work as it may be called asynchronously; you need to track whether this is call has occurred and then return a promise or this value from line 1028 if required